lesson17/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const textBox = document.createElement("p");
+  textBox.classList.add("");
+  textBox.textContent = error;
+  slideImageArea.appendChild(textBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  document.querySelector(".mainvisual__images").appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) sliderItem.classList.add("is-active");
```

Comment:
```suggestion
    i === 0 && sliderItem.classList.add("is-active");
```
nits
It can also be written without writing if.

Reply:
You can omit it by using the AND operator (&&).
I learned this!

[3fb7f97](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/3fb7f9753292de91bf06889a0b807f2f26a8cf33)